### DISCUSSION TITLE
add callback support for `defaultValidationErrorsShape`

### DISCUSSION THREAD

**Discussion by mammadataei** [next-safe-action/next-safe-action#302] [Ideas]
I need to change the validation errors shape globally, but `defaultValidationErrorsShape` is not dynamic. A transformed callback like `handleValidationErrorsShape` but in createSafeActionClient would be a great option.

**Comment by TheEdoRan** [comment]
Yes, I agree, it would be great! But unfortunately it's impossible to infer the validation errors structure types from a client-level function. That's why you can only set "formatted" or "flattened" at the instance level.

**Reply by mammadataei** [reply]
What about providing a generic validation errors type to the client instance?

```typescript
type ValidationErrors = ...

createSafeActionClient<ValidationErrors>({
  defaultValidationErrorsShape: (ve): ValidationErrors => {}
})
```
